float32 ball_pos_x
float32 ball_pos_y
float32 ball_radius